tests/unittests/Foundation/NSSetTests.mm

```diff
@@ -148,4 +148,98 @@
+    NSString* expected = @"{(\n"
+                          "        {(\n"
+                          "                (\n"
+                          "            2,\n"
+                          "            3,\n"
+                          "                        {(\n"
+                          "                4\n"
+                          "            )}\n"
+                          "        ),\n"
+                          "        1,\n"
+                          "        5\n"
+                          "    )},\n"
+                          "    6\n"
+                          ")}";
+    EXPECT_OBJCEQ(expected, nestedSet.description);
+}
+
+TEST(NSSet, MakeObjectsPerformSelector) {
+    NSSet* set = [NSSet setWithObjects:@[ @1, @2, @3 ], [NSSet setWithObjects:@4, @5, nil], @[ @6 ], nil];
+    __block NSMutableSet* otherSet = [NSMutableSet set];
+    [set makeObjectsPerformSelector:@selector(enumerateObjectsUsingBlock:)
+                         withObject:^(id obj, BOOL* stop) {
+                             [otherSet addObject:obj];
+                         }];
+
+    EXPECT_EQ(6, otherSet.count);
+    EXPECT_TRUE([otherSet containsObject:@1]);
+    EXPECT_TRUE([otherSet containsObject:@2]);
+    EXPECT_TRUE([otherSet containsObject:@3]);
+    EXPECT_TRUE([otherSet containsObject:@4]);
+    EXPECT_TRUE([otherSet containsObject:@5]);
+    EXPECT_TRUE([otherSet containsObject:@6]);
+}
+
+TEST(NSSet, ObjectsWithOptionsPassingTest) {
+    NSSet* set = [NSSet setWithObjects:@1, @2, @3, @4, @5, nil];
+
+    // Verify that the NSEnumerationConcurrent option executes the blocks concurrently
+    __block NSCondition* condition = [[NSCondition new] autorelease];
+    __block unsigned waitingCount = 0;
+    NSSet* matchingForward = [set objectsWithOptions:NSEnumerationConcurrent
+                                         passingTest:^BOOL(id obj, BOOL* stop) {
+                                             [condition lock];
+                                             waitingCount++;
+                                             if (waitingCount == 5) {
+                                                 waitingCount--;
+                                                 [condition signal];
+                                                 [condition unlock];
+                                                 return [obj unsignedIntegerValue] % 2 == 0 ? YES : NO;
+                                             }
+
+                                             [condition wait];
+                                             waitingCount--;
+                                             [condition signal];
+                                             [condition unlock];
+                                             return [obj unsignedIntegerValue] % 2 == 0 ? YES : NO;
+                                         }];
+
+    ASSERT_EQ(0, waitingCount);
+
+    NSSet* matchingReverse = [set objectsWithOptions:(NSEnumerationConcurrent | NSEnumerationReverse)
```

Comment:
I see what you're saying. You're just using the reverse enumeration as another set to match against. I still think it shouldn't be used though, since it is still technically undefined.